src/utils/CommonUtils.js

```diff
@@ -277,7 +301,45 @@ class CommonUtils {
 			.replace(/>/g, '&gt;')
 			.replace(/"/g, '&quot;')
 			.replace(/'/g, '&apos;');
+	}
+
+	/**
+	 * Retries a function before finally failing
+	 * @param func {function}
+	 * @param retries {number|null}
+	 * @param waitOnError {boolean}
+	 * @returns {*} funtion result
+	 * @throws {string} final fail if all retries expired
+	 */
+	static async retry(func, retries = 5, waitOnError = true) {
+		let error = "";
+		const sleep = require('util').promisify(setTimeout);
+
+		for (let i = 1; i <= retries; ++i) { // retry
+			try {
+				return await func();
+			} catch (e) {
+				const time = CommonUtils.exponentialTimeWithJitter(i);
+				error = e;
+				console.warn(`Call failed with error '${error}'. Retry [${i}/${retries}]` + (i < retries && waitOnError ? ` waiting ${time}ms` : ""));
+				if (i < retries && waitOnError) await sleep(time);
+			}
+		}
+		throw error;
+	}
 
+	static async requestAsync(opts) {
+		return new Promise((resolve, reject) => {
+			const cb = (err, response, body) => {
+				if(err) {
+					reject(err);
+				} else {
+					resolve([response, body]);
+				}
+			};
+			const request = require('request');
+			request(opts, cb);
+		});
```

Comment:
maybe something like `(req, cb) => request(req, (err, response, body) => cb(err, [response, body]))` and promisify that?